src/internal/obj/mem_client.go

```diff
@@ -0,0 +1,90 @@
+package obj
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+)
+
+type memClient struct {
```

Comment:
We discussed this offline, but I just want to make a note about going ahead and moving to the key / value store model for the purposes of the in-memory cache. We just need a simple in-memory, LRU cache that stores the plaintext chunks. This would just be used by the chunk storage layer.

Reply:
Also, we wouldn't need the void client.

Reply:
These changes are done.